src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReduceToCollectionCommand.cs

```diff
@@ -145,16 +145,30 @@ private BlittableJsonReaderObject GenerateReduceOutput(object reduceObject, Inde
 
             using (_patternForReduceOutputReferences?.BuildReferenceDocumentId(out referenceDocIdBuilder))
             {
+                var patternPropertiesAddedSuccessfully = true;
+
                 foreach (var property in _index.OutputReduceToCollectionPropertyAccessor.GetPropertiesInOrder(reduceObject))
                 {
                     var value = property.Value;
                     djv[property.Key] = TypeConverter.ToBlittableSupportedType(value, context: _indexContext);
 
                     if (referenceDocIdBuilder?.ContainsField(property.Key) == true)
-                        referenceDocIdBuilder.Add(property.Key, property.Value);
+                    {
+                        try
+                        {
+                            referenceDocIdBuilder.Add(property.Key, property.Value);
+                        }
+                        catch (Exception e)
+                        {
+                            patternPropertiesAddedSuccessfully = false;
+
+                            if (stats != null) // should never be null
+                                stats.AddReduceError($"Failed to build document ID based on provided pattern for output to collection references. {e.Message}");
```

Comment:
Can we add the full exception details?

Reply:
Fixed in https://github.com/ravendb/ravendb/pull/10285/commits/14b715f1b78381ca8b839ca027e8fd7c57e4ee2e